src/ducks/apps/components/Sections/index.spec.jsx

```diff
@@ -3,111 +3,85 @@
   })
 
   it('should render correctly render message if error provided', () => {
-    const mockOnAppClick = jest.fn()
-    const component = shallow(
-      <Sections
-        t={tMock}
-        lang="en"
-        subtitle="Test Apps"
-        apps={mockApps}
-        allApps={mockApps}
-        onAppClick={mockOnAppClick}
-        error={new Error('This is a test error')}
-        location={{ search: '' }}
-      />
-    ).getElement()
-    expect(component).toMatchSnapshot()
+    const { root } = setup({
+      props: { error: new Error('This is a test error') }
+    })
+    expect(() => root.getByText('This is a test error')).not.toThrow()
   })
+})
 
-  it('should not render dropdown filter on mobile if nav=false flag provided', () => {
-    const mockOnAppClick = jest.fn()
-    const component = shallow(
-      <Sections
-        t={tMock}
-        lang="en"
-        subtitle="Test Apps"
-        apps={mockApps}
-        allApps={mockApps}
-        onAppClick={mockOnAppClick}
-        breakpoints={{ isMobile: true }}
-        hasNav={false}
-      />
-    ).getElement()
-    expect(component).toMatchSnapshot()
-  })
+const sleep = duration => new Promise(resolve => setTimeout(resolve, duration))
 
-  it('should not render dropdown filter on tablet if nav=false flag provided', () => {
-    const mockOnAppClick = jest.fn()
-    const component = shallow(
-      <Sections
-        t={tMock}
-        lang="en"
-        subtitle="Test Apps"
-        apps={mockApps}
-        allApps={mockApps}
-        onAppClick={mockOnAppClick}
-        breakpoints={{ isTablet: true }}
-        hasNav={false}
-      />
-    ).getElement()
-    expect(component).toMatchSnapshot()
+describe('Search', () => {
+  beforeEach(() => {
+    flag('store.search', true)
   })
 
-  it('should render dropdown filter on mobile if no nav=false flag provided', () => {
-    const mockOnAppClick = jest.fn()
-    const component = shallow(
-      <Sections
-        t={tMock}
-        lang="en"
-        subtitle="Test Apps"
-        apps={mockApps}
-        allApps={mockApps}
-        onAppClick={mockOnAppClick}
-        breakpoints={{ isMobile: true }}
-      />
-    ).getElement()
-    expect(component).toMatchSnapshot()
+  afterEach(() => {
+    flag('store.search', false)
   })
 
-  it('should render dropdown filter on tablet if no nav=false flag provided', () => {
-    const mockOnAppClick = jest.fn()
-    const component = shallow(
-      <Sections
-        t={tMock}
-        lang="en"
-        subtitle="Test Apps"
-        apps={mockApps}
-        allApps={mockApps}
-        onAppClick={mockOnAppClick}
-        breakpoints={{ isTablet: true }}
-      />
-    ).getElement()
-    expect(component).toMatchSnapshot()
+  it('should filter the results', async () => {
+    const { root } = setup()
+    const input = root.getByPlaceholderText('"SNCF", "telecom", "bills"')
+    act(() => {
+      fireEvent.change(input, { target: { value: 'Bouil' } })
+    })
+    await sleep(500)
```

Comment:
Why 500? Because of the debounce (now at 300) ?

Reply:
Yes. I changed the code to use a mocked debounce to circumvent this problem (so that the tests are faster). Jest fake timers do not work with debounce unfortunately.

Reply:
I love the idea of mocking the debounce! <3